src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6613,6 +6613,19 @@ private StatementSyntax ParsePossibleDeclarationOrBadAwaitStatement(ref ResetPoi
         /// </remarks>
         private StatementSyntax ParseStatementNoDeclaration(bool allowAnyExpression)
         {
+            if (this.CurrentToken.ContextualKind == SyntaxKind.AwaitKeyword)
```

Comment:
Why do this at the top instead of handle it in the big switch statement below, like we do for expression statements that start with await?